apps/app/src/views/routes/Order.tsx

```diff
@@ -85,11 +121,73 @@ export const Order = () => {
   }
 
   const isMobile = useBreakpointValue({ base: true, sm: false });
-  const tableWidth = useBreakpointValue({ base: 'full', sm: '100%', md: '75%' });
+  const rightColWidth = '75%';
+  const { colorMode } = useColorMode();
+
+  const CopyText = ({ text }: { text: string }) => {
+    if (!text) return null;
+    return (
+      <HStack spacing={2} justifyContent={isMobile ? 'end' : 'start'}>
+        <Text
+          fontSize="md"
+          whiteSpace={isMobile ? 'nowrap' : undefined}
+          overflow={isMobile ? 'hidden' : undefined}
+          textOverflow={isMobile ? 'ellipsis' : undefined}
+          maxWidth={isMobile ? '100px' : undefined}
+        >
+          {text}
+        </Text>
+        <IconButton
+          variant="ghost"
+          color="gray.500"
+          aria-label="Copy external id"
+          minW="fit-content"
+          h="fit-content"
+          py={0}
+          my={0}
+          _hover={{ backgroundColor: 'transparent' }}
+          icon={<FiCopy size="1.3em" />}
+          onClick={() => navigator.clipboard.writeText(text || '')}
+        />
+      </HStack>
+    );
+  };
 
   return (
     <Page kicker="Order" header={order ? formatFills(order.fills) : ''} loading={loading}>
       <VStack spacing={4} fontSize={{ base: 'md', md: 'lg' }} alignItems="start" w="100%" mt={0}>
+        {loading ? (
+          <Skeleton width="112px" height="32px" borderRadius="md" />
+        ) : (
+          <Button
+            size="sm"
+            aria-label="Cancel Order"
+            isDisabled={order.fulfillment?.type !== types.FulfillmentType.MailOrder}
+            onClick={async () => {
+              const decision = await confirmWrapper('Cancel this order?', {
+                description: 'You will not be able to recover this order.',
```

Comment:
Feel like we can modify the copy a little, maybe "You will not be able to undo this" or something?

Reply:
updated